packages/node_modules/@ciscospark/internal-plugin-metrics/test/unit/spec/metrics.js

```diff
@@ -93,9 +119,30 @@ describe(`plugin-metrics`, () => {
     describe(`#submitClientMetrics()`, () => {
       describe(`before login`, () => {
         it(`posts pre-login metric`, () => {
-          metrics.submitClientMetrics(eventName, mockPayload, preLoginId);
-          assert.calledWith(metrics.postPreLoginMetric, preLoginProps, preLoginId);
-          assert.calledOnce(spark.credentials.getClientToken);
+          const date = Date.now();
+          const promise = metrics.submitClientMetrics(eventName, mockPayload, preLoginId);
+          return promiseTick(50)
+            .then(() => clock.tick(config.metrics.batcherWait))
+            .then(() => assert.isFulfilled(promise))
```

Comment:
don't use isFulfilled. Tran is working on removing chai-as-promised because the latest version will never be compatible with Internet Explorer. just returning the promise should work.